libs/dal/src/shared/interceptors/cached.interceptor.ts

```diff
@@ -0,0 +1,52 @@
+import { buildCachedQuery, buildKey, CacheInterceptorTypeEnum } from './shared-cache';
+
+// eslint-disable-next-line @typescript-eslint/naming-convention
+export function Cached(storeKeyPrefix?: string) {
+  return (target: any, key: string, descriptor: any) => {
+    const originalMethod = descriptor.value;
+    const methodName = key;
+
+    descriptor.value = async function (...args: any[]) {
+      const skip = args[2]?.skip;
+      if (!this.cacheService?.cacheEnabled() || skip) return await originalMethod.apply(this, args);
+
+      const query = buildCachedQuery(args);
+
+      const cacheKey = buildKey(
+        storeKeyPrefix ?? this.MongooseModel.modelName,
+        methodName,
+        query,
+        CacheInterceptorTypeEnum.CACHED
+      );
+
+      if (!cacheKey) {
+        return await originalMethod.apply(this, args);
+      }
+
+      try {
+        const value = await this.cacheService.get(cacheKey);
+        if (value) {
+          return JSON.parse(value);
+        }
+      } catch (err) {
+        // eslint-disable-next-line no-console
+        console.error(`An error has occurred when extracting "key: ${key}`, 'CacheInterceptor', err);
```

Comment:
Being the cache service a sensitive element I think we would need to raise any error (maybe Sentry?) instead of shadowing them.

Reply:
Agree this one was a temporary solution need to check what is the best way to inject logger instance to this service.